client/reader/site-stream/featured.jsx

```diff
@@ -49,24 +80,29 @@ export default React.createClass( {
 		}
 	},
 
-	handleClick( post ) {
+	handleClick( postData ) {
+		let post = postData.post;
 		stats.recordTrack( 'calypso_reader_clicked_featured_post', { blog_id: post.site_ID, post_id: post.ID } )
 		stats.recordAction( 'clicked_featured_post' );
 		stats.recordGaEvent( 'Clicked Featured Post' );
 
-		page( '/read/post/id/' + post.site_ID + '/' + post.ID );
+		let url;
+		if ( postData.isDiscoverPost ) {
+			url = '/read/post/id/' + postData.originalPost.site_ID + '/' + postData.originalPost.ID;
+		} else {
+			url = '/read/post/id/' + post.site_ID + '/' + post.ID;
+		}
```

Comment:
> This is arguable

Not arguable - I _actually do_ think the ternary is easier to read 😉

But that's fine if you think the `if` is easier :smile: